homework/list.js

```diff
@@ -1,79 +1,83 @@
 'use strict';
 
-(function() {
+(function(exports) {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+  var ListManager = function() {
+    var _listNoteContent = [];
+    var _wrapper = null;  };
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+  ListManager.prototype = {
+    start() {
+      this._wrapper = document.querySelector('#note-list-wrapper');
+      this.fetchList().then((function(data) {
+        this.updateList(data);
+        this.drawList();
+        this.preloadFirstNote();
+      }).bind(this));
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+      window.addEventListener('click', (function(event) {
```

Comment:
While you use binding this in the previous `addEventListener`, you should keep the convention consistent. And, It's a convention to prefix method with `on` for the corresponding function, so it should be renamed as 'onClick'.